sympy/stats/joint_rv_types.py

```diff
@@ -404,21 +404,24 @@ class MultinomialDistribution(JointDistribution):
     is_Discrete = True
 
     def check(self, n, p):
-        _value_check(((n > 0) != False) and isinstance(n, Integer),
+        _value_check(((n > 0) != False),
                         "number of trials must be a positve integer")
         for p_k in p:
-            _value_check((p_k >= 0) != False,
+            _value_check((p_k >= 0) != False and (p_k <= 1) != False,
                         "probability must be at least a positive symbol.")
+        _value_check(Eq(sum(p), 1) != False,
+                        "probabilities must sum to 1")
 
     @property
     def set(self):
-        return Range(0, self.n)**len(self.p)
+        i = Symbol('i', negative=False, integer=True)
+        return imageset(i, i, Interval(0, self.n))**len(self.p)
```

Comment:
Should this be `imageset(i, i, Intersection(S.Naturals0, Interval(0, self.n)))**len(self.p)` ? BTW: I'm not sure that assumptions make any difference. When doing this Intersection, it will try to create a Range -- something should be modified so returns the imageset as-is when working with symbols.
```
>>> imageset(i, i, Intersection(Interval(x,y), S.Naturals0))
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/sets/sets.py", line 1213, in __new__
    return simplify_intersection(args)
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/sets/sets.py", line 1966, in simplify_intersection
    new_set = intersection_sets(s, t)
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/multipledispatch/dispatcher.py", line 198, in __call__
    return func(*args, **kwargs)
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/sets/handlers/intersection.py", line 20, in intersection_sets
    return Intersection(S.Integers, b, Interval(a._inf, S.Infinity))
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/sets/sets.py", line 1213, in __new__
    return simplify_intersection(args)
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/sets/sets.py", line 1966, in simplify_intersection
    new_set = intersection_sets(s, t)
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/multipledispatch/dispatcher.py", line 198, in __call__
    return func(*args, **kwargs)
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/sets/handlers/intersection.py", line 31, in intersection_sets
    s = Range(ceiling(b.left), floor(b.right) + 1)
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/sets/fancysets.py", line 431, in __new__
    [0, 1/10, 1/5].'''))
ValueError: 
Finite arguments to Range must be integers; `imageset` can define
other cases, e.g. use `imageset(i, i/10, Range(3))` to give [0, 1/10, 1/5].
```

Reply:
What happens if you return None after catching ValueError at /sets/handlers/intersection.py", line 31?

Reply:
@smichr The change you suggested makes the things work for now.